### PR TITLE
added vs by oid method

### PR DESCRIPTION
**Pull request by daco101**
This adds a method to the cql_measure model that returns value sets by oid, used in the backend calculation.

Pull requests into Bonnie Bundler require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] JIRA ticket for this PR: https://jira.mitre.org/browse/BONNIE-1530
- [x] JIRA ticket links to this PR
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [x] Code coverage has not gone down and all code touched or added is covered.
    * In rare situations, this may not be possible or applicable to a PR. In those situations:
        1. Note why this could not be done or is not applicable here:
        2. Add TODOs in the code noting that it requires a test
        3. Add a JIRA task to add the test and link it here:
- [x] Automated regression test(s) pass (N/A, adding a method does not affect calculation in any way --Luke)

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA test links: NA
- [x] Justification for using JIRA tests: NA
- [x] JIRA tests have been added to sprint NA

**Reviewer 1:**

Name: @losborne
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass NA
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree NA

**Reviewer 2:**

Name: @jbradl11 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

### CODE REVIEW

File: lib/models/cql_measure.rb [hunk 1]
```diff
@@ -72,6 +72,23 @@ def value_sets
...
 
+  #note, this method could be duplicate onto the measure model as well
+  def value_sets_by_oid
+    @value_sets_by_oid = {}
+    value_sets.each do |vs|
+      if @value_sets_by_oid[vs.oid]
+        # if there are multiple value sets with the same oid for this user, then keep the one with
+        # the version corresponding to this measure for the fixture.
+        if vs.version.include?(hqmf_set_id)
+          @value_sets_by_oid[vs.oid] = { vs.version => vs }
```
Comment (lukoktonos) [inline lib/models/cql_measure.rb]: It looks like the test never hits this line.  I wrote a loop on bonnie to see which measures from `bonnie-fixtures@mitre.org` might hit it, and here's which ones got to that condition:

CMS158v6
CMS160v6
CMS177v6
CMS32v7
CMS146v6
CMS136v7
CMS720v0
CMS759v1
CMS722v0
CMS760v0
CMS52v7
CMS761v0
CMSv9999
CMS529v0
CMS334v1
CMS460v0

Here's the loop if it's useful:
```
    task :all_value_sets_by_oid => :environment do
      bonnie_fixtures_user_id = User.find_by email: 'bonnie-fixtures@mitre.org'
      bonnie_fixture_measures = CqlMeasure.where user_id: bonnie_fixtures_user_id  
      bonnie_fixture_measures.all.each do |measure|
        if measure.value_sets_by_oid
          puts measure.cms_id
        end
      end
```

**Review by lukoktonos — CHANGES_REQUESTED** [review]

**Review by jbradl11 — CHANGES_REQUESTED** [review]
This should be into a different branch. Create a bonnie-v2.3-backend-calculation branch (based off of Bonnie-v2.3) and make a PR into that. We want to leave Bonnie-v2.3 alone until the entire feature is complete.

File: lib/models/cql_measure.rb [hunk 2]
```diff
@@ -72,6 +72,23 @@ def value_sets
     @value_sets
   end
 
+  #note, this method could be duplicate onto the measure model as well
```
Comment (jbradl11) [inline lib/models/cql_measure.rb]: You can remove this note. Measure model is no longer used.

File: lib/models/cql_measure.rb [hunk 3]
```diff
@@ -72,6 +72,23 @@ def value_sets
...
+  def value_sets_by_oid
+    @value_sets_by_oid = {}
+    value_sets.each do |vs|
+      if @value_sets_by_oid[vs.oid]
+        # if there are multiple value sets with the same oid for this user, then keep the one with
+        # the version corresponding to this measure for the fixture.
```
Comment (jbradl11) [inline lib/models/cql_measure.rb]: Looks like this comment needs to be updated.

**Comment by jbradl11** [comment]
The comment still needs to be updated (it mentions fixtures).

**Review by jbradl11 — APPROVED** [review]
Change looks good.

**Review by lukoktonos — APPROVED** [review]
